src/models/blockchain/block/mod.rs

```diff
@@ -2017,260 +1829,745 @@ pub(crate) mod tests {
+            //             b_new.proof = super::super::BlockProof::SingleProof(Proof::invalid());
+            //             // b_new.kernel.appendix = BlockAppendix::new(BlockAppendix::consensus_claims(b_new.body()));
+            //             b_new.validate(&b_prev, ts, Network::Main).await
+            //         });
+            //         prop_assert_eq![BlockValidationError::ProofValidity, t.err().unwrap()];
+            //     }
+            // );
+            let err = BlockValidationError::MutatorSetUpdateIntegrity;
+            proptest::proptest!(
+                deterministic,
+                |((mut b_prev, ts, rness) in n_strategy(), proof in arb::<NeptuneProof>(),
+                record_addition_an in arb::<AdditionRecord>())|
+                // tx_kernel in crate::tests::shared::strategies::txkernel::default(true))|
+                {
+                    // fake_block_successor_with_merged_tx
+
+                    dbg!("`record_addition_an`");
+                    let t = crate::tests::tokio_runtime().block_on(async {
+                        let b_new = fake_valid_successor_for_tests(
+                            &b_prev,
+                            ts,
+                            rness,
+                            Network::Main,
+                        ).await;
+
+                        // dbg!(b_new.proof);
+                        // b_new.kernel.appendix = BlockAppendix::new(BlockAppendix::consensus_claims(b_new.body()));
+
+                        // let mut tx_kernel_bad_update = TransactionKernelProxy::from(b_prev.kernel.body.transaction_kernel.clone());
+                        // tx_kernel_bad_update.inputs = Vec::new();
+                        // tx_kernel_bad_update.outputs = [record_addition_an; 2].to_vec();
+                        // tx_kernel_bad_update.body.mutator_set_accumulator.add(&record_addition_an);
+                        b_prev.kernel.body.mutator_set_accumulator.add(&record_addition_an);
+                        // b_prev.kernel.body.transaction_kernel = tx_kernel_bad_update.into_kernel();
+                        // // MutatorSetUpdate{additions: [record_addition_an; 2].to_vec(), removals: Vec::new()}.apply_to_accumulator(&mut b_now_bad_update.kernel.body.mutator_set_accumulator);
+                        b_prev.kernel.appendix = BlockAppendix::new(BlockAppendix::consensus_claims(b_prev.body()));
+                        // // dbg!(&b_now_bad_update.kernel.body.transaction_kernel.inputs);
+
+                        b_new.validate(&b_prev, ts, Network::Main).await
+                    });
+                    prop_assert_eq![err, t.err().unwrap()];
+                }
             );
-            tx_kernel_big_coinbase.coinbase = Some(
-                NativeCurrencyAmount::one()
-                    + Block::block_subsidy(b_now_big_coinbase.header().height),
+            err_variants.retain(|v| v != &err);
+
+            let err = BlockValidationError::RemovalRecordsValidity;
```

Comment:
The git commit hook disallows unrelated variable shadowing. (And CI will fail if you commit with `-n` anyway.) As `err` is only used once between every binding, you might as well inline the value. That would also benefit readability IMHO.